### DISCUSSION TITLE
Add a setup page to tweak stuff without editing the text file

### DISCUSSION THREAD

**Discussion by Onosendaifr** [probonopd/MiniDexed#682] [Ideas]
Some choices should be done directly on a page without the need to extract SD to edit the .ini file.

For exemple "Engine Type", "Sound Device", "Program Change Enable" etc...

**Comment by probonopd** [comment]
Well. The settings that are not hardware setup related should be accessible via the menu.

But everything that has to do with the hardware (e.g., pinout, display, sound device, etc.) is intentionally not part of the menu, so that you can't misconfigure the setup by using the menu, effectively locking you out.

Engine Type, Program Change Enable could be added to the menu imho. Are there others?

**Reply by miotislucifugis** [reply]
Midi thru configuration/ behavior should be a performance setting.   Also setting the default Performance - "performance to load"

Maybe for midi, which is semi hardware related, there could be the hardware config setting in the .ini, which declare what hardware is actually present, and the menu options for configuring the midi routing would reflect those hardware limitations.   for example, if there was no serial midi in/out circuit, the menu would only offer USB routing options.

**Comment by Onosendaifr** [comment]
Thanks !

Others :

- Program Change mode
- Program Change action

MIDI Button Navigation should also be doable via a menu with a MIDI learn feature
